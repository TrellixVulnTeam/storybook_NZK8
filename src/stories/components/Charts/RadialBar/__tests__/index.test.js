import React from 'react';
import { mount } from 'enzyme';
import UpkRadialBarChart from 'components/Charts/RadialBar/index';

describe('<UpkRadialBarChart />', () => {
  const mandatoryProps = {
    data: [
      {
        name: 'Name',
        value: 0,
        fill: 'red',
      },
    ],
  };

  it('should render correctly with mandatory props', () => {
    const sut = mount(<UpkRadialBarChart {...mandatoryProps} />);

    expect(sut).toMatchSnapshot();
  });

  it(`should render correctly with full props and window's width 1036`, () => {
    const props = {
      ...mandatoryProps,
      percentValue: true,
    };
    const sut = mount(<UpkRadialBarChart {...props} />);

    expect(sut).toMatchSnapshot();
  });
});
