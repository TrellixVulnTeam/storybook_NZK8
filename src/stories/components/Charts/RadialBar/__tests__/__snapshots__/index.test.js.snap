// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpkRadialBarChart /> should render correctly with full props and window's width 1036 1`] = `
<UpkRadialBarChart
  data={
    Array [
      Object {
        "fill": "red",
        "name": "Name",
        "value": 0,
      },
    ]
  }
  percentValue={true}
>
  <styled__ChartWrapper
    height="300px"
  >
    <div
      className="styled__ChartWrapper-sc-18uhklm-0 cumIDA"
      height="300px"
    >
      <ResponsiveContainer
        debounce={0}
        height="100%"
        width="100%"
      >
        <div
          className="recharts-responsive-container"
          style={
            Object {
              "height": "100%",
              "maxHeight": undefined,
              "minHeight": undefined,
              "minWidth": undefined,
              "width": "100%",
            }
          }
        >
          <RadialBarChart
            barCategoryGap="10%"
            barGap={4}
            barSize={28}
            cx="50%"
            cy="50%"
            data={
              Array [
                Object {
                  "fill": "red",
                  "name": "Name",
                  "value": 0,
                },
              ]
            }
            endAngle={-30}
            height={0}
            innerRadius="70%"
            layout="radial"
            margin={
              Object {
                "bottom": 5,
                "left": 5,
                "right": 5,
                "top": 5,
              }
            }
            outerRadius="100%"
            reverseStackOrder={false}
            stackOffset="none"
            startAngle={210}
            width={0}
          />
          <ResizeDetector
            handleHeight={true}
            handleWidth={true}
            onResize={[Function]}
            refreshRate={1000}
            resizableElementId=""
            skipOnMount={false}
          >
            <div
              style={
                Object {
                  "display": "none",
                  "height": 0,
                  "position": "absolute",
                  "visibility": "hidden",
                  "width": 0,
                }
              }
            />
          </ResizeDetector>
        </div>
      </ResponsiveContainer>
    </div>
  </styled__ChartWrapper>
</UpkRadialBarChart>
`;

exports[`<UpkRadialBarChart /> should render correctly with mandatory props 1`] = `
<UpkRadialBarChart
  data={
    Array [
      Object {
        "fill": "red",
        "name": "Name",
        "value": 0,
      },
    ]
  }
>
  <styled__ChartWrapper
    height="300px"
  >
    <div
      className="styled__ChartWrapper-sc-18uhklm-0 cumIDA"
      height="300px"
    >
      <ResponsiveContainer
        debounce={0}
        height="100%"
        width="100%"
      >
        <div
          className="recharts-responsive-container"
          style={
            Object {
              "height": "100%",
              "maxHeight": undefined,
              "minHeight": undefined,
              "minWidth": undefined,
              "width": "100%",
            }
          }
        >
          <RadialBarChart
            barCategoryGap="10%"
            barGap={4}
            barSize={28}
            cx="50%"
            cy="50%"
            data={
              Array [
                Object {
                  "fill": "red",
                  "name": "Name",
                  "value": 0,
                },
              ]
            }
            endAngle={-30}
            height={0}
            innerRadius="70%"
            layout="radial"
            margin={
              Object {
                "bottom": 5,
                "left": 5,
                "right": 5,
                "top": 5,
              }
            }
            outerRadius="100%"
            reverseStackOrder={false}
            stackOffset="none"
            startAngle={210}
            width={0}
          />
          <ResizeDetector
            handleHeight={true}
            handleWidth={true}
            onResize={[Function]}
            refreshRate={1000}
            resizableElementId=""
            skipOnMount={false}
          >
            <div
              style={
                Object {
                  "display": "none",
                  "height": 0,
                  "position": "absolute",
                  "visibility": "hidden",
                  "width": 0,
                }
              }
            />
          </ResizeDetector>
        </div>
      </ResponsiveContainer>
    </div>
  </styled__ChartWrapper>
</UpkRadialBarChart>
`;
